File: problems/Continuous Subarray Sum/checkSubarraySum.cs
public class Solution {
    public bool CheckSubarraySum(int[] nums, int k) {
        var store = new Dictionary<int, int> {{0, 0}};
        var sum = 0;

        for (var i = 0; nums.Length > i; ++i) {
            sum += nums[i];

            if (0 != k) {
                sum %= k;
            }

            if (!store.ContainsKey(sum)) {
                store.Add(sum, 1 + i);
            } else if (store[sum] < i) {
                return true;
            }
        }

        return false;
    }
}

File: problems/Continuous Subarray Sum/checkSubarraySum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> sumToIdxMap = new HashMap<>();
        int sum = 0;

        sumToIdxMap.put(0, 0);

        for (int idx = 0; nums.length > idx; ++idx) {
            sum += nums[idx];
            sum %= k;

            if (!sumToIdxMap.containsKey(sum)) {
                sumToIdxMap.put(sum, 1 + idx);
            } else if (0 < idx - sumToIdxMap.get(sum)){
                return true;
            }
        }

        return false;
    }
}
